### PR TITLE
Previously missing PB subtypes added

### PR DESCRIPTION
**Pull request by sangeetashukla**
### Purpose/implementation Section
Add PPTID subtypes

#### What scientific question is your analysis addressing?
There were some PPT ID subtypes that were not added to the module earlier. The module needed updates to include those additional subtypes.

#### What was your approach?
Review the input data source to ascertain which subtypes were missed.
Correctly map the pathology_free_text_diagnosis to molecular_subtype_methyl
Followed by mapping molecular_subtype correctly
Run the module to re-generate subtypes in the results/ directory

#### What GitHub issue does your pull request address?
[Issue 590](https://github.com/d3b-center/OpenPedCan-analysis/issues/590)

#### Which areas should receive a particularly close look?
Please review the subtypes for their acceptable nomenclature

#### Is there anything that you want to discuss further?
No

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd [hunk 1]
```diff
@@ -46,7 +46,14 @@ pb_meta <- histo %>%
     dkfz_v12_methylation_subclass == "PIN_RB" ~ "PB, RB1",
     dkfz_v12_methylation_subclass == "PB_GRP1A" ~ "PB, Group 1A",
     dkfz_v12_methylation_subclass == "PB_GRP1B" ~ "PB, Group 1B",
-    dkfz_v12_methylation_subclass == "PB_GRP2" ~ "PB, Group 2")) 
+    dkfz_v12_methylation_subclass == "PB_GRP2" ~ "PB, Group 2",
+    dkfz_v12_methylation_subclass == "ETMR_C19MC" ~ "EMTR, C19MC",
+    dkfz_v12_methylation_subclass == "PTPR_B" ~ "PTPR, Beta",
+    dkfz_v12_methylation_subclass == "PPTID_A" ~ "PPTID, Alpha",
+    dkfz_v12_methylation_subclass == "MB_WNT" ~ "MB, WNT",
+    dkfz_v12_methylation_subclass == "CTRL_PIN" ~ "CTRL, PIN"
```
Comment (jharenza) [inline analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd]:
```suggestion
    dkfz_v12_methylation_subclass == "PB_GRP2" ~ "PB, Group 2",
    dkfz_v12_methylation_subclass == "PPTID_A" ~ "PB, PPTID",
```
hi @sangeetashukla I think we only wanted to add this one - can you update ?

File: analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd [hunk 2]
```diff
@@ -62,14 +69,28 @@ pb_meta <- histo %>%
 ## PB_GRP2: pineoblastoma, miRNA processing-altered 2 (PB, Group 2)
 ## PIN_RB: pineoblastoma, RB1-altered (PB, RB1)
 
+## Additional subtypes added on 07/22/2024
+## Edited By: Sangeeta Shukla
+## ETMR_C19MC: Embryonal tumor with multilayered rosettes (ETMR), C19MC-altered
+## PTPR_B: Protein Tyrosine Phosphatase Receptor-Type Beta
+## PPTID_A: Ppineal Parenchymal Tumor of Intermediate Differentiation Type A
+## MB_WNT: Medulloblastoma, WNT-activated
+## CTRL_PIN: Prostatic Intraepithelial Neoplasia (PIN), Control
```
Comment (jharenza) [inline analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd]:
```suggestion
## PPTID_A: Ppineal Parenchymal Tumor of Intermediate Differentiation Type A
```

File: analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd [hunk 3]
```diff
@@ -62,14 +69,28 @@ pb_meta <- histo %>%
...
   mutate(molecular_subtype = case_when(molecular_subtype_methyl == "PB, MYC/FOXR2" ~ "PB, MYC/FOXR2", 
                                        molecular_subtype_methyl == "PB, RB1" ~ "PB, RB1",
                                        molecular_subtype_methyl %in% c("PB, Group 1A", "PB, Group 1B") ~ "PB, Group 1",
-                                       molecular_subtype_methyl == "PB, Group 2" ~ "PB, Group 2"))
+                                       molecular_subtype_methyl == "PB, Group 2" ~ "PB, Group 2",
+                                       molecular_subtype_methyl == "EMTR, C19MC" ~ "EMTR, C19MC",
+                                       molecular_subtype_methyl == "PTPR, Beta" ~ "PTPR, Beta",
+                                       molecular_subtype_methyl == "PPTID, Alpha" ~ "PPTID, Alpha",
+                                       molecular_subtype_methyl == "MB, WNT" ~ "MB, WNT",
+                                       molecular_subtype_methyl == "CTRL, PIN" ~ "CTRL, PIN"))
```
Comment (jharenza) [inline analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd]:
```suggestion
                                       molecular_subtype_methyl == "PB, Group 2" ~ "PB, Group 2",
                                       molecular_subtype_methyl == "PB, PPTID" ~ "PB, PPTID"))
```

**Review by jharenza — CHANGES_REQUESTED** [review]
hi @sangeetashukla I left some comments for changes required here

**Comment by sangeetashukla** [comment]
> hi @sangeetashukla I left some comments for changes required here

Hi @jharenza I have made the changes. Please take a look when you can.

File: analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd [hunk 4]
```diff
@@ -62,14 +65,21 @@ pb_meta <- histo %>%
 ## PB_GRP2: pineoblastoma, miRNA processing-altered 2 (PB, Group 2)
 ## PIN_RB: pineoblastoma, RB1-altered (PB, RB1)
 
+## Additional subtypes added on 07/22/2024
+## Edited By: Sangeeta Shukla
+## PPTID_A: Ppineal Parenchymal Tumor of Intermediate Differentiation Type A
```
Comment (jharenza) [inline analyses/molecular-subtyping-PB/01-molecular-subtype-pineoblastoma.Rmd]:
```suggestion
## PPTID_A: Pineal Parenchymal Tumor of Intermediate Differentiation Type A
```

**Review by jharenza — APPROVED** [review]
hi @sangeetashukla thanks for the update. I found a bug where the low conf methyl subtypes were being propagated and pushed a commit for that. This looks good now, no new samples subtyped but the code exists now. Let's see if it passes.

**Pull request merged**